src/sage/groups/braid.py

```diff
@@ -1657,6 +1718,82 @@ def is_conjugated(self, other):
         """
         l = conjugatingbraid(self, other)
         return bool(l)
+    
+    def pure_conjugating_braid(self, other):
+        r"""
+        Return a pure conjugating braid, if it exists.
+
+        INPUT:
+
+        - ``other`` -- the other braid to look for conjugating braid
+
+        EXAMPLES::
+
+            sage: B = BraidGroup(4)
+            sage: a = B([1, 2, 3])
+            sage: b = B([3, 2,])
+            sage: c = b ^ 12 * a / b ^ 12
+            sage: d1 = a.conjugating_braid(c)
+            sage: print (len(d1.Tietze()))
+            30
+            sage: print (d1.permutation())
+            [3, 4, 1, 2]
+            sage: d1 * c / d1 == a
+            True
+            sage: d1 * a / d1 == c
+            False
+            sage: d2 = a.pure_conjugating_braid(c)
+            sage: print (len(d2.Tietze()))
+            24
+            sage: print (d2.permutation())
+            [1, 2, 3, 4]
+            sage: d2 * c / d2 == a
+            True
+            sage: print (d2)
+            (s0*s1*s2^2*s1*s0)^4
+            sage: print(a.conjugating_braid(b))
+            None
+            sage: print(a.pure_conjugating_braid(b))
+            None
+            sage: a1=B([1])
+            sage: a2=B([2])
+            sage: a1.conjugating_braid(a2)
+            s1*s0
+            sage: a1.permutation()
+            [2, 1, 3, 4]
+            sage: a2.permutation()
+            [1, 3, 2, 4]
+            sage: print (a1.pure_conjugating_braid(a2))
+            None
+            sage: (a1^2).conjugating_braid(a2^2)
+            s1*s0
+            sage: print ((a1^2).pure_conjugating_braid(a2^2))
+            None
+        """
+        B = self.parent()
+        n = B.strands()
+        G = SymmetricGroup(n)
+        p1 = G(self.permutation())
+        p2 = G(other.permutation())
+        if p1 != p2:
+            return None
+        b0 = self.conjugating_braid(other)
+        if not b0:
+            return None
+        p3 = G(b0.permutation().inverse())
+        if p3.is_one():
+            return b0
+        LB = self.centralizer()
+        LP = [G(a.permutation()) for a in LB]
+        if p3 not in G.subgroup(LP):
+            return None
+        P = p3.word_problem(LP, display = False, as_list = True)
+        b1 = prod(LB[LP.index(G(a))] ** b for a,b in P)
```

Comment:
What happens if `P` is empty? Add a doctest checking for this corner case (if it can occur).

Reply:
Additionally, if you make `LP` a `dict` from permutations to braids, this becomes faster and you can still pass it off to `G.subgroup()`.

Reply:
For the first question a previous `if` makes impossible to have `P` empty. I could not pass the dict to `subgroup` and even if it would be possible, I think I cannot pass it to `word_problem`

Reply:
> For the first question a previous `if` makes impossible to have `P` empty.

I might say also the previous `p3.is_one()` also makes that the case. However, again this case should have a doctest.

> I could not pass the dict to `subgroup` and even if it would be possible, I think I cannot pass it to `word_problem`

You can pass a `dict` to `subgroup`:
```python
sage: G = groups.permutation.Symmetric(5)
sage: d = {G.random_element(): i for i in range(5)}
sage: d
{(1,5,2,4): 0, (1,5)(2,3): 1, (1,3,5,2): 2, (1,5,3,2): 3, (1,5)(2,4,3): 4}
sage: G.subgroup(d)
Subgroup generated by [(1,3,5,2), (1,5,3,2), (1,5)(2,3), (1,5)(2,4,3), (1,5,2,4)] of (Symmetric group of order 5! as a permutation group)
```

Indeed for `word_problem`, you cannot pass the `dict`, but you can do `list(d)`, which will have a list in the same order the `dict` `d` was constructed (Python3 guarantees this).